Firestore/core/include/firebase/firestore/timestamp.h

```diff
@@ -205,18 +205,16 @@ std::chrono::time_point<Clock, Duration> Timestamp::ToTimePoint() const {
       chr::duration_cast<Duration>(chr::nanoseconds(nanoseconds_));
   return TimePoint{seconds + nanoseconds};
 }
-
 #endif  // !defined(_STLPORT_VERSION)
 
 }  // namespace firebase
 
 namespace std {
 template <>
 struct hash<firebase::Timestamp> {
-  size_t operator()(const firebase::Timestamp& timestamp) const {
-    return 37 * static_cast<size_t>(timestamp.seconds()) +
-           static_cast<size_t>(timestamp.nanoseconds());
-  }
+  // Note: specialization of `std::hash` is provided for convenience only. The
+  // implementation is not cryptographically secure and is subject to change.
```

Comment:
I don't think anyone expects specializations of `std::hash` to be cryptographically secure. The subject to change part is good though.